components/brave_wallet_ui/common/slices/endpoints/network.endpoints.ts

```diff
@@ -24,11 +23,43 @@ export const NETWORK_TAG_IDS = {
   SWAP_SUPPORTED: 'SWAP_SUPPORTED'
 } as const
 
+interface IsEip1559ChangedMutationArg {
+  id: string
+  isEip1559: boolean
+}
+
 export const networkEndpoints = ({
   mutation,
   query
 }: WalletApiEndpointBuilderParams) => {
   return {
+    // queries
+    /** Gets all networks, regardless of which coin-types are enabled */
+    getAllKnownNetworks: query<BraveWallet.NetworkInfo[], void>({
+      queryFn: async (_arg, { endpoint }, extraOptions, baseQuery) => {
+        try {
+          const { data: api } = baseQuery(undefined)
+
+          const { networks: ethNetworks } =
+            await api.jsonRpcService.getAllNetworks(BraveWallet.CoinType.ETH)
+          const { networks: solNetworks } =
+            await api.jsonRpcService.getAllNetworks(BraveWallet.CoinType.SOL)
+          const { networks: filNetworks } =
+            await api.jsonRpcService.getAllNetworks(BraveWallet.CoinType.FIL)
+
```

Comment:
I think I'll have to refactor backend as we don't actually need per coin networks but just 'all networks'. This will also cover cases for missing BTC and ZEC here